app/src/main/java/study_dev/testbed/studyhopper/ui/groupFinder/StudyGroupFinderFragment.java

```diff
@@ -0,0 +1,202 @@
+                R.array.age_range_preference, android.R.layout.simple_spinner_item);
+        ageRangeAdapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item);
+        ageRangeSpinner.setAdapter(ageRangeAdapter);
+        //groupPreferencesSpinner.setOnItemSelectedListener(this);
+
+        ArrayAdapter<CharSequence> genderAdapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.group_preferences, android.R.layout.simple_spinner_item);
+        genderAdapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item);
+        genderSpinner.setAdapter(genderAdapter);
+
+        ArrayAdapter<CharSequence> locationAdapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.location_preference, android.R.layout.simple_spinner_item);
+        locationAdapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item);
+        locationSpinner.setAdapter(locationAdapter);
+
+        ArrayAdapter<CharSequence> groupSizeAdapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.group_size_preference, android.R.layout.simple_spinner_item);
+        groupSizeAdapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item);
+        groupSizeSpinner.setAdapter(groupSizeAdapter);
+    }
+
+    private void getGroupsFromFirebase() {
+        CollectionReference groupsRef = db.collection("groups");
+
+        Query query = buildQuery(groupsRef);
+
+        query.get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                if (task.isSuccessful()) {
+                    groupList.clear();
+                    for (QueryDocumentSnapshot document : task.getResult()) {
+                        Group group = document.toObject(Group.class);
+
+                        groupListItem listItem = new groupListItem(group.getGroupName(),
+                                group.getCourseCode(),
+                                group.getGroupColor());
+
+                        groupList.add(listItem);
+                    }
+                    reloadRecycler();
+                } else {
+                    Log.d(TAG, "Error getting documents: ", task.getException());
+                }
+            }
+        });
+    }
+
+    private void reloadRecycler() {
+        if (groupList.size() == 0) {
+            noResultsText.setVisibility(View.VISIBLE);
+        } else {
+            noResultsText.setVisibility(View.GONE);
+            mAdapter = new GroupSearchAdapter(groupList);
+            mGroupRecycleView.setAdapter(mAdapter);
+        }
+    }
+
+    private Query buildQuery(CollectionReference groupsReference) {
+        Query newQuery = groupsReference.whereEqualTo("isActive", true);
+
+        String courseSubject = subjectSearch.getText().toString();
```

Comment:
Ensure there is validation for course subjects and codes. As it stands I can type in a bunch of garbage and on multiple lines. Lets utilize 3 char characters for the prefix  and then a four digit number. Regex or simple java validation will due.